core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -298,6 +298,23 @@
     assertThat(root, hasTree(expected));
   }
 
+  @Test public void testTableFunctionScan() {
+    // Equivalent SQL:
+    //   SELECT *
+    //   FROM TABLE(DEDUP(CURSOR(select * from emp), CURSOR(select * from DEPT), 'NAME'))
+    final RelBuilder builder = RelBuilder.create(config().build());
+    RelNode root = builder.scan("EMP").scan("DEPT")
+        .functionScan(new MockSqlOperatorTable.DedupFunction(),
+            Lists.newArrayList(builder.cursor(0),
+                builder.cursor(1))).build();
+    final String expected = "LogicalTableFunctionScan(invocation="
+        + "[DEDUP(CURSOR($0), CURSOR($1))], "
+        + "rowType=[RecordType(VARCHAR(1024) NAME)])\n"
+        + "  LogicalTableScan(table=[[scott, EMP]])\n"
+        + "  LogicalTableScan(table=[[scott, DEPT]])\n";
+    assertThat(root, hasTree(expected));
+  }
```

Comment:
Add an assert that builder's stack is now empty. (functionScan should have popped both the EMP and DEPT scans, though I suspect it doesn't currently.)

I think the best way to check is to call builder.build() and make sure that it throws. Or perhaps call builder.toString(), and make sure that it is empty.

Reply:
Got it. Will do.